docs/source/configuration/advanced_configuration.md

```diff
@@ -262,58 +262,73 @@ Since both of the file names (`catalog.yml` and `catalog_globals.yml`) match the
+### How to use resolvers in the `OmegaConfigLoader`
+Instead of hard-coding values in your configuration files, you can also dynamically compute them using `OmegaConf`'s
+[resolvers](https://omegaconf.readthedocs.io/en/2.3_branch/custom_resolvers.html#resolvers) functionality. You use resolvers to define custom
+logic to calculate values of parameters or catalog entries, or inject these values from elsewhere. To use this feature with Kedro, you can pass a
+`dict` of custom resolvers to `OmegaConfigLoader` through `CONFIG_LOADER_ARGS` in your project's `src/<package_name>/settings.py`.
 The example below illustrates this:
 
 ```python
+import polars as pl
+from datetime import date
+
 from kedro.config import OmegaConfigLoader
-from omegaconf import OmegaConf
-from typing import Any, Dict
-
-
-class CustomOmegaConfigLoader(OmegaConfigLoader):
-    def __init__(
-        self,
-        conf_source: str,
-        env: str = None,
-        runtime_params: Dict[str, Any] = None,
-    ):
-        super().__init__(
-            conf_source=conf_source, env=env, runtime_params=runtime_params
-        )
-
-        # Register a customer resolver that adds up numbers.
-        self.register_custom_resolver("add", lambda *numbers: sum(numbers))
-
-    @staticmethod
-    def register_custom_resolver(name, function):
-        """
-        Helper method that checks if the resolver has already been registered and registers the
-        resolver if it's new. The check is needed, because omegaconf will throw an error
-        if a resolver with the same name is registered twice.
-        Alternatively, you can call `register_new_resolver()` with  `replace=True`.
-        """
-        if not OmegaConf.has_resolver(name):
-            OmegaConf.register_new_resolver(name, function)
-```
 
-In order to use this custom configuration loader, you will need to set it as the project configuration loader in `src/<package_name>/settings.py`:
+CONFIG_LOADER_CLASS = OmegaConfigLoader
 
-```python
-from package_name.custom_configloader import CustomOmegaConfigLoader
 
-CONFIG_LOADER_CLASS = CustomOmegaConfigLoader
-```
+def date_today():
+    return date.today()
 
-You can then use the custom "add" resolver in your `parameters.yml` as follows:
 
+CONFIG_LOADER_ARGS = {
+    "custom_resolvers": {
+        "add": lambda *my_list: sum(my_list),
+        "polars": lambda x: getattr(pl, x),
+        "today": lambda: date_today(),
+    }
+}
+```
+These custom resolvers are then registered using `OmegaConf.register_new_resolver()` under the hood and can be used in any of the
+configuration files in your project. For example, you can use the `add` or the `today` resolver defined above in your `parameters.yml` like this:
 ```yaml
 model_options:
-  test_size: ${add:1,2,3}
+  test_size: "${add:1,2,3}"
```

Comment:
Is this necessary? What happen if you don't string quote it.

Reply:
Yeah, otherwise you get a `ParserError: Invalid YAML or JSON file <filepath>, unable to read line 3, position 23.`